prac_06/programming_language.py

```diff
@@ -0,0 +1,36 @@
+class ProgrammingLanguage:
+    """Represent information about a programming language."""
+
+    def __init__(self, name, typing, reflection, year):
+        """Construct a ProgrammingLanguage from the given values."""
+        self.name = name
+        self.typing = typing
+        self.reflection = reflection
+        self.year = year
+
+    def __str__(self):
+        """Return string representation of a ProgrammingLanguage."""
+        return f"{self.name}, {self.typing} Typing, Reflection={self.reflection}, First appeared in {self.year}"
+
+    def is_dynamic(self):
+        """Determine if language is dynamically typed."""
+        return self.typing == "Dynamic"
+
+
+def run_tests():
+    """Run simple tests/demos on ProgrammingLanguage class."""
+    ruby = ProgrammingLanguage("Ruby", "Dynamic", True, 1995)
+    python = ProgrammingLanguage("Python", "Dynamic", True, 1991)
+    visual_basic = ProgrammingLanguage("Visual Basic", "Static", False, 1991)
+
+    languages = [ruby, python, visual_basic]
+    print(python)
+
+    print("The dynamically typed languages are:")
+    for language in languages:
+        if language.is_dynamic():
+            print(language.name)
```

Comment:
Consider adding an additional docstring here to clarify what is being tested in this for loop.